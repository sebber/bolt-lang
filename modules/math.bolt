fun add(a: Integer, b: Integer): Integer {
    return a + b
}

fun multiply(x: Integer, y: Integer): Integer {
    return x * y
}

fun square(n: Integer): Integer {
    return n * n
}

export add
export multiply
export square