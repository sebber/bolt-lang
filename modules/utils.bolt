fun max(a: Integer, b: Integer): Integer {
    if (a > b) {
        return a
    } else {
        return b
    }
}

fun min(a: Integer, b: Integer): Integer {
    if (a < b) {
        return a
    } else {
        return b
    }
}

export max
export min