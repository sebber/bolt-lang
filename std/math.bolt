export fun abs(x: Integer): Integer {
    if (x < 0) {
        return 0 - x
    } else {
        return x
    }
}

export fun min(a: Integer, b: Integer): Integer {
    if (a < b) {
        return a
    } else {
        return b
    }
}

export fun max(a: Integer, b: Integer): Integer {
    if (a > b) {
        return a
    } else {
        return b
    }
}