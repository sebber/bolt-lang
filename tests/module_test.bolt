import { add, multiply } from "modules/math"
import { max } from "modules/utils"

val result1 := add(5, 3)
val result2 := multiply(4, 7)
val result3 := max(result1, result2)

print(result1)
print(result2)
print(result3)

print("Module test complete!")