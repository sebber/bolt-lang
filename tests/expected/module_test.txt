8
28
28
Module test complete!